stimela/task_stats.py

```diff
@@ -236,6 +238,22 @@ def update_stats(now: datetime, sample: TaskStatsDatum):
         start = _task_start_time.setdefault(key, now)
         _taskstats[key][0] = (now - start).total_seconds()
 
+def update_children():
+    """Update the module level dictionary mapping child pid to process.
+
+    This is necessary as calling Process.children will return different
+    Process objects each time. These then fail to report CPU stats unless
+    we make them block which has a large impact on performance.
+    """
+    current_children = stimela_process.children(recursive=True)
+    current_pids = {proc.pid for proc in current_children}
+    child_processes.update(
```

Comment:
Isn't that the same as ``child_processes.update(current_children)``, or am I missing something?

Reply:
NVM -- I am missing something -- it's a dict not a set.